cobbler/modules/managers/isc.py

```diff
@@ -47,367 +48,281 @@ def __init__(self, api: "CobblerAPI"):
+        blend_data = utils.blender(self.api, False, system)
+
+        system_config = self._gen_system_config(system, blend_data, distro)
+        self.config = utils.merge_dicts_recursive(
+            self.config,
+            {"dhcp_tags": system_config},
+        )
+        self.config["date"] = time.asctime(time.gmtime())
+        self._write_configs(self.config)
+        return self.restart_service()
+
+    def remove_single_system(self, system_obj: "System") -> None:
+        if not self.config:
+            self.write_configs()
+            return
+
+        profile: Optional["Profile"] = system_obj.get_conceptual_parent()  # type: ignore
+        distro: Optional["Distro"] = profile.get_conceptual_parent()  # type: ignore
+        blend_data = utils.blender(self.api, False, system_obj)
+
+        system_config = self._gen_system_config(system_obj, blend_data, distro)
+        for dhcp_tag in system_config:
+            for mac_address in system_config[dhcp_tag]:
+                self.config.get("dhcp_tags", {}).get(dhcp_tag, {}).pop(mac_address, "")
+        self.config["date"] = time.asctime(time.gmtime())
+        self._write_configs(self.config)
+        self.restart_service()
 
-        blender_cache: Dict[str, Any] = {}
-
-        with open(template_file, "r", encoding="UTF-8") as template_fd:
-            template_data = template_fd.read()
-
-        # Use a simple counter for generating generic names where a hostname is not available.
-        counter = 0
+    def _gen_system_config(
+        self,
+        system_obj: "System",
+        system_blend_data: Dict[str, Any],
+        distro_obj: Optional["Distro"],
+    ) -> Dict[str, Any]:
+        """
+        Generate DHCP config for a single system.
 
-        # We used to just loop through each system, but now we must loop through each network interface of each system.
+        :param system_obj: System to generate DHCP config for
+        :param system_blend_data: utils.blender() data for the System
+        :param distro_object: Optional, is used to access distro-specific information like arch when present
+        """
         dhcp_tags: Dict[str, Any] = {"default": {}}
-
-        # FIXME: ding should evolve into the new dhcp_tags dict
-        ding: Dict[str, Any] = {}
-        ignore_macs: List[str] = []
-
-        for system in self.systems:
-            if not system.is_management_supported(cidr_ok=False):
+        processed_system_master_interfaces: Set[str] = set()
+        ignore_macs: Set[str] = set()
+        if not system_obj.is_management_supported(cidr_ok=False):
+            self.logger.debug(
+                "%s does not meet precondition: MAC, IPv4, or IPv6 address is required.",
+                system_obj.name,
+            )
+            return {}
+
+        profile: Optional["Profile"] = system_obj.get_conceptual_parent()  # type: ignore
+        for iface_name, iface_obj in system_obj.interfaces.items():
+            # TODO: move to own function
```

Comment:
Just wondering, are we taking now the opportunity to do this as well?

Reply:
I wasn't planning on it, but let me try :)

Reply:
Actually, after taking a look, it doesn't make much sense to me. I can move the code, but it doesn't really simplify reading the function. It just moves 90% of the function into a different function, which doesn't really help with readability or understanding.

WDYT? I can probably just remove the comment.

Reply:
Ok, sure. Let's then remove the comment. Thanks!